packages/react-app/src/components/pages/Bounties/index.tsx

```diff
@@ -32,6 +38,78 @@ const FilterResultPlaceholder = ({ message }: { message: string }): JSX.Element
 	</Stack>
 );
 
+const SelectExport = (({ bounties, selectedBounties, setSelectedBounties }: {
+	bounties: BountyCollection[] | undefined,
+	selectedBounties: string[],
+	setSelectedBounties: SetState<string[]>,
+	}): JSX.Element => {
+
+	const { colorMode } = useColorMode();
+	const [csvData, setCsvData] = useState<BountyCollection[]>([]);
+	const csvLink = useRef<CSVLink & HTMLAnchorElement & { link: HTMLAnchorElement }>(null);
+
+	const handleSelectAll = (): void => {
+		if (bounties && selectedBounties.length < bounties.length) {
+			setSelectedBounties(bounties.map(({ _id }) => _id));
+		}
+	};
+
+	const handleClearAll = (): void => {
+		setSelectedBounties([]);
+	};
+
+	const handleCSV = (): void => {
+		if (bounties && csvLink.current) {
+			const getCsvData = bounties.filter(item => selectedBounties.includes(item._id)).map(b => MiscUtils.csvEncode(b) as BountyCollection);
+			setCsvData(getCsvData);
+			// Force state to update before invoking the download
+			setTimeout(() => {
+				csvLink?.current?.link.click();
+			});
```

Comment:
Do you know of a better way to handle this? Turns out the state wasn't updating by the time the CSVLink was clicked. The setTimeout() hack makes sure the process finishes before the click.

Reply:
Set timeout hack would indicate to me you have a problem with operation ordering inside the event loop. I'll take a look though

Reply:
Try this:

```ts
	const handleCSV = (): void => {
		if (bounties && csvLink.current) {
			const getCsvData = bounties
				.filter(({ _id }) => selectedBounties.includes(_id))
				.map(MiscUtils.csvEncode);
				
			setCsvData(() => getCsvData);
			csvLink?.current?.link.click();
		}
	};
```
Also, I don't think CSVData should be of type BountyCollection[], it is a stringified object of type `Record<string, string>;`

Reply:
https://stackoverflow.com/questions/42038590/when-to-use-react-setstate-callback

Reply:
Getting unresolved errors on T and C in here. Do I need to import something?

type UseStateCallbackReturn = [T, (state: T, cb: C) => void];

Reply:
Oh yeah, needs the generics to be defined (I separated it in the example for readability)

If you replace the return type of the function (`UseSetStateCallbackReturn`) with the actual type definition, it will accept T and C from the generics.

Reply:
Edited back to the original, sorry about that